openquake/hazardlib/gsim/parker_2020.py

```diff
@@ -0,0 +1,752 @@
+    """)
+
+    CONSTANTS = {"b4": 0.1, "f3": 0.05, "Vb": 200,
+                 "vref_fnl": 760, "V1": 270, "vref": 760}
+
+
+class ParkerEtAl2020SSlab(ParkerEtAl2020SInter):
+    """
+    Modifications for subduction slab.
+    """
+
+    DEFINED_FOR_TECTONIC_REGION_TYPE = const.TRT.SUBDUCTION_INTRASLAB
+
+    # slab also requires hypo_depth
+    REQUIRES_RUPTURE_PARAMETERS = {'mag', 'hypo_depth'}
+
+    def _c0(self, C, C_PGA):
+        """
+        c0 factor.
+        """
+        if self.saturation_region is None:
+            c0_col = "c0slab"
+        elif self.region in ["AK", "SA"]:
+            c0_col = self.saturation_region + "_c0slab"
+        else:
+            # no more specific region available
+            c0_col = self.region + "_c0slab"
+        return C[c0_col], C_PGA[c0_col]
+
+    def _get_mb(self):
+        """
+        Regional Mb factor.
+        """
+        mb_regions = ("Aleutian", "AK", "Cascadia", "CAM_S", "CAM_N",
+                      "JP_Pac", "JP_Phi", "SA_N", "SA_S", "TW_W", "TW_E")
+        if self.saturation_region in mb_regions:
+            i = mb_regions.index(self.saturation_region)
+            m_b = (7.98, 7.2, 7.2, 7.6, 7.4, 7.65,
+                   7.55, 7.3, 7.25, 7.7, 7.7)[i]
+        else:
+            m_b = 7.6
+
+        return m_b
+
+    def _suffix(self):
+        """
+        Column name constant suffix based on class.
+        """
+        return "slab"
+
+    def _path_term_h(self, mag, m_b=None):
+        """
+        H factor for path term, subduction slab.
+        """
+        if mag <= m_b:
+            m = (math.log10(35) - math.log10(3.12)) / (m_b - 4)
+            return 10 ** (m * (mag - m_b) + math.log10(35))
+        return 35
+
+    def _a0(self, C, C_PGA, region):
+        """
+        Regional anelastic coefficient for subduction slab, a0
+        """
+        if self.region is None:
+            return C["a0slab"], C_PGA["a0slab"]
+        return C[self.region + "_a0slab"], C_PGA[self.region + "_a0slab"]
+
+    def _depth_scaling(self, C, hypo_depth):
+        if hypo_depth >= C["db"]:
+            return C["d"]
+        if hypo_depth <= 20:
+            return C["m"] * (20 - C["db"]) + C["d"]
+        return C["m"] * (hypo_depth - C["db"]) + C["d"]
+
+
+class ParkerEtAl2020SInterAleutian(ParkerEtAl2020SInter):
```

Comment:
All of these sub-classes seem redundant to me. The only difference between them seems to be the default parameters. From what I can see, the main ParkerEtAl2020SInter and ParkerEtAlSSlab are supporting the different regions, so these two classes should be sufficient to cover all of the uses